modules/fedimint-ln/src/contracts/outgoing.rs

```diff
@@ -30,7 +34,21 @@ pub struct Preimage(pub [u8; 32]);
 impl IdentifyableContract for OutgoingContract {
     fn contract_id(&self) -> ContractId {
         let mut engine = ContractId::engine();
-        Encodable::consensus_encode(self, &mut engine).expect("Hashing never fails");
+        Encodable::consensus_encode(&self.hash, &mut engine).expect("Hashing never fails");
+        Encodable::consensus_encode(&self.gateway_key, &mut engine).expect("Hashing never fails");
+        Encodable::consensus_encode(&self.timelock, &mut engine).expect("Hashing never fails");
+        Encodable::consensus_encode(&self.user_key, &mut engine).expect("Hashing never fails");
+        Encodable::consensus_encode(&self.invoice, &mut engine).expect("Hashing never fails");
```

Comment:
The contract id should not change when we switch the refund flag. The hashing did indeed not change, but since I added a field I had to be more explicit about which fields to hash.